### PR TITLE
docs(samples): Add dataflow_batch_write_to_storage snippet

### PR DESCRIPTION
**Pull request by VeronicaWasson**
## Description

Add code snippet for writing to Cloud Storage from Dataflow

For this doc: https://cloud.google.com/dataflow/docs/guides/write-to-cloud-storage

Also set up snippets folder for Dataflow

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dataflow/snippets/tests/__init__.py
Comment (msampathkumar) [inline dataflow/snippets/tests/__init__.py:1]: This file is missing a copyright header. Also this file may not required!

File: dataflow/snippets/__init__.py
Comment (msampathkumar) [inline dataflow/snippets/__init__.py:1]: This file is missing a copyright header. Also this file may not required!

**Review by m-strzelczyk — APPROVED** [review]
Please update the year in your license headers.

File: dataflow/snippets/tests/__init__.py
Comment (VeronicaWasson) [reply dataflow/snippets/tests/__init__.py:1]: Removing these causes this error:

```
ImportError while importing test module '/workspace/dataflow/snippets/tests/test_batch_write_storage.py'.
Hint: make sure your test modules/packages have valid Python names.
```

(I got the same error running the tests locally without the __init__ files)

Comment (VeronicaWasson) [reply dataflow/snippets/tests/__init__.py:1]: Based on this, I left in these files but added the copyright header

File: dataflow/snippets/batch_write_storage.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# [START dataflow_batch_write_to_storage]
+import apache_beam as beam
+from apache_beam.io.textio import WriteToText
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+def write_to_cloud_storage():
```
Comment (msampathkumar) [inline dataflow/snippets/batch_write_storage.py:22]: How does this function know the output location or bucket name? As per test-cases, some arguments are sent to `sys.argv`, but this function does not seem to parse any sys.args

Passing the bucket name as an argument to this `write_to_cloud_storage(bucket_name: str):`

File: dataflow/snippets/tests/test_batch_write_storage.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        yield
+    finally:
+        bucket.delete(force=True)
+
+
+def test_write_to_cloud_storage(setup_and_teardown):
+    sys.argv = ['', f'--output=gs://{bucket_name}/output/out-']
+    write_to_cloud_storage()
+
+    blobs = list(storage_client.list_blobs(bucket_name))
```
Comment (msampathkumar) [inline dataflow/snippets/tests/test_batch_write_storage.py:41]: As per `write_to_cloud_storage`, output format is `.txt` files. Can you add a test-case to ensure that `.txt` is generated ?

File: dataflow/snippets/batch_write_storage.py (hunk 1 above)
Comment (VeronicaWasson) [reply dataflow/snippets/batch_write_storage.py]: This is idiomatic for Beam in Python. For example, [here](https://github.com/GoogleCloudPlatform/dataflow-cookbook/blob/main/Python/gcs/read_avro.py). The `PipelineOptions` class automatically parses the arguments. I added a code comment to try and clarify.

File: dataflow/snippets/tests/test_batch_write_storage.py (hunk 2 above)
Comment (VeronicaWasson) [reply dataflow/snippets/tests/test_batch_write_storage.py:41]: oh good idea, thanks. Added.

Comment (VeronicaWasson) [reply dataflow/snippets/tests/test_batch_write_storage.py:41]: I put it in the same test, should it be a separate test?

Comment (msampathkumar) [reply dataflow/snippets/tests/test_batch_write_storage.py:41]: you can use the same test case.

File: dataflow/snippets/batch_write_storage.py (hunk 1 above)
Comment (msampathkumar) [reply dataflow/snippets/batch_write_storage.py]: Thank you.

**Review by msampathkumar — APPROVED** [review]

**Comment by msampathkumar** [comment]
@VeronicaWasson Approved with a suggestion.

**Pull request merged**